Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -223,6 +232,32 @@ public class PBXProjGenerator {
             return Set(files.map { $0.buildFile.reference })
         }
 
+        func getRunScript(runScript: RunScript) throws -> PBXShellScriptBuildPhase {
+
+            let shellScript: String
+            switch runScript.script {
+            case let .path(path):
+                shellScript = try getPath(path).read()
+            case let .script(script):
+                shellScript = script
+            }
+            let escapedScript = shellScript.replacingOccurrences(of: "\"", with: "\\\"").replacingOccurrences(of: "\n", with: "\\n")
+            let shellScriptPhase = PBXShellScriptBuildPhase(
+                reference: generateUUID(PBXShellScriptBuildPhase.self, String(describing: runScript.name) + shellScript),
+                files: [],
+                name: runScript.name ?? "Run Script",
+                inputPaths: Set(runScript.inputFiles),
+                outputPaths: Set(runScript.outputFiles),
+                shellPath: runScript.shell ?? "/bin/sh",
```

Comment:
Yeah that would be an option. I've been trying to keep any specific knowledge out of ProjectSpec though, and just keep it dumb. 
Also I think that `/bin/sh` should be the default value in the `xcodeproj` constructor, and I didn't want to double up work there. It's something we still need to add to xcodeproj though